src/Sarif.UnitTests/Visitors/MakeUriAbsoluteVisitorTest.cs

```diff
@@ -103,79 +98,117 @@ public void MakeUriAbsoluteVisitor_VisitPhysicalLocation_DoesNotSetUriIfBaseIsNo
+            expectedRun.ValueEquals(actualRun).Should().BeTrue();
         }
 
         [Fact]
         public void MakeUriAbsoluteVisitor_VisitSarifLog_MultipleRunsWithDifferentProperties_RebasesProperly()
         {
-            Run runA = GenerateRunForTest(new Dictionary<string, Uri>()
+            Run runA = GenerateRunForTest(new Dictionary<string, FileLocation>()
             {
-                { "%TEST1%", new Uri(@"C:\srcroot\") },
-                { "%TEST2%", new Uri(@"D:\bld\out\") },
+                ["%TEST1%"] = new FileLocation { Uri = new Uri(@"C:\srcroot\") },
+                ["%TEST2%"] = new FileLocation { Uri = new Uri(@"D:\bld\out\") }
             });
-            Run runB = GenerateRunForTest(new Dictionary<string, Uri>()
+            Run runB = GenerateRunForTest(new Dictionary<string, FileLocation>()
             {
-                { "%TEST1%", new Uri(@"C:\src\abc") },
-                { "%TEST2%", new Uri(@"D:\bld\123\") },
+                ["%TEST1%"] = new FileLocation { Uri = new Uri(@"C:\src\abc\") },
+                ["%TEST2%"] = new FileLocation { Uri = new Uri(@"D:\bld\123\") }
             });
             MakeUrisAbsoluteVisitor visitor = new MakeUrisAbsoluteVisitor();
 
-            SarifLog log = new SarifLog() { Runs=new Run[] { runA, runB } };
+            SarifLog log = new SarifLog() { Runs = new Run[] { runA, runB } };
             SarifLog newLog = visitor.VisitSarifLog(log);
 
             // Validate
             newLog.Runs.Should().HaveCount(2);
             newLog.Runs[0].Files.Should().NotIntersectWith(newLog.Runs[1].Files);
         }
+
+        [Fact]
+        public void MakeUriAbsoluteVisitor_CombineUriFunctionsProperly()
+        {
+            var testCases = new Tuple<string, string, string>[]
+            {
+                new Tuple<string, string, string>
+                    (@"https://base/", @"relative/file.cpp",  "https://base/relative/file.cpp")
+            };
+
+            foreach (Tuple<string, string, string> testCase in testCases)
+            {
+                MakeUrisAbsoluteVisitor.CombineUris(
+                    absoluteBaseUri: new Uri(testCase.Item1, UriKind.Absolute),
+                    relativeUri: new Uri(testCase.Item2, UriKind.Relative))
+                        .Should().Be(testCase.Item3);
+            }
+        }
+
+        [Fact]
+        public void MakeUriAbsoluteVisitor_CombineUriValidatesArgumentsProperly()
+        {
+            Uri absoluteUri = new Uri("https://absolute.example.com", UriKind.Absolute);
+            Uri relativeUri = new Uri("relative/someResource", UriKind.Relative);
+
+            // First, ensure that our test data succeeds when used properly
+            MakeUrisAbsoluteVisitor.CombineUris(
+                absoluteBaseUri: absoluteUri,
+                relativeUri: relativeUri);
+            
+            // Pass relative uri where absolute expected
+            var action = new Action(() => 
```

Comment:
>Action [](start = 29, length = 6)

There's a more compact syntax for this.

Reply:
Fixed.

---
In reply to: [247331689](https://github.com/Microsoft/sarif-sdk/pull/1210#discussion_r247331689) [](ancestors = 247331689)